### ISSUE TITLE
[ban-types] Include extendDefaults in doc's example configuration

### ISSUE THREAD

**Issue by jzabinski** [typescript-eslint/typescript-eslint#2222]
**Repro**

```JSON
{
  "rules": {
    "@typescript-eslint/ban-types": [
            "error",
            {
                "extendsDefaults": false
            }
        ]
  }
}
```

```TS
Lint any code using this configuration
```

**Expected Result**
Configuration for rule "@typescript-eslint/ban-types" is invalid
**Actual Result**
Configuration for rule "@typescript-eslint/ban-types" is invalid

**Additional Info**
After getting the above error, I discovered that the problem was that I was trying to use the property 'extendsDefaults', not 'extendDefaults'. That took a long time to find. Could the [documentation](https://github.com/typescript-eslint/typescript-eslint/blob/master/packages/eslint-plugin/docs/rules/ban-types.md) be updated to include the property in the example configuration so that this mistake is harder to make?

**Versions**

| package                            | version |
| ---------------------------------- | ------- |
| `@typescript-eslint/eslint-plugin` | `2.27.0` |
| `@typescript-eslint/parser`        | `2.27.0` |
| `TypeScript`                       | `3.8.3` |
| `ESLint`                           | `6.8.0` |
| `node`                             | `12.13.1` |
| `npm`                              | `6.13.4` |

**Comment by bradzacher** [comment]
The option is documented in the `options` section - is this not enough?
https://github.com/typescript-eslint/typescript-eslint/blob/master/packages/eslint-plugin/docs/rules/ban-types.md#options

Also, there isn't an example configuration really, just a json blob which shows the different styles of configuration.

**Comment by jzabinski** [comment]
I tend to copy the example as-is when implementing rules, adjusting the configuration as needed. In this approach, sometimes it is hard to see how exactly a property should appear if the property is added manually (both its name and the correct '}' under which to insert the property). Shame on me for not just copy/pasting the property from the Options object; but it does seem like it could prevent a headache in the future.

**Comment by bradzacher** [comment]
Most of our rules don't even have example configurations, only the more complex ones.

As mentioned, that particular example is only there to show that you can use a string or an object.

We generally want to avoid example config, as they are another set of code to maintain; when the rule changes you then have to update the config.

When we do include an example, we include the minimum amount of configuration for that example - this is to save people adding extra, unnecessary props into their config. That way if we want to remove a property in future, it reduces the potential surface area and thus reduces major version upgrade pain for users.

In this instance, the `extendDefaults` option would be superfluous, so it's better to omit it.
